cli/crates/cli/src/login.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{errors::CliError, output::report};
+use backend::{login, types::LoginMessage};
+use common::utils::get_thread_panic_message;
+use indicatif::{ProgressBar, ProgressStyle};
+use std::{
+    sync::mpsc::{channel, RecvTimeoutError},
+    thread::{sleep, spawn},
+    time::Duration,
+};
+
+pub fn login() -> Result<(), CliError> {
+    let (message_sender, message_receiver) = channel();
+
+    let login_handle = spawn(|| login::login(message_sender).map_err(CliError::BackendError));
+
+    if let Ok(LoginMessage::CallbackUrl(url)) = message_receiver.recv() {
```

Comment:
`LoginMessage::CallbackUrl(url)` has to be the first message from `login`, debating structuring this differently